packages/react/src/components/ComposedModal/next/ModalFooter.js

```diff
@@ -0,0 +1,252 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import Button from '../../Button';
+import ButtonSet from '../../ButtonSet';
+import classNames from 'classnames';
+import { usePrefix } from '../../../internal/usePrefix';
+
+function SecondaryButtonSet({
+  secondaryButtons,
+  secondaryButtonText,
+  secondaryClassName,
+  closeModal,
+  onRequestClose,
+}) {
+  function handleRequestClose(evt) {
+    closeModal(evt);
+    onRequestClose(evt);
+  }
+
+  const secondaryClass = classNames({
+    [secondaryClassName]: secondaryClassName,
+  });
+
+  if (Array.isArray(secondaryButtons) && secondaryButtons.length <= 2) {
+    return secondaryButtons.map(({ buttonText, onClick: onButtonClick }, i) => (
+      <Button
+        key={`${buttonText}-${i}`}
+        className={secondaryClass}
+        kind="secondary"
+        onClick={onButtonClick || handleRequestClose}>
+        {buttonText}
+      </Button>
+    ));
+  }
+  if (secondaryButtonText) {
+    return (
+      <Button
+        className={secondaryClass}
+        onClick={handleRequestClose}
+        kind="secondary">
+        {secondaryButtonText}
+      </Button>
+    );
+  }
+  return null;
+}
+
+SecondaryButtonSet.propTypes = {
+  closeModal: PropTypes.func,
+  onRequestClose: PropTypes.func,
+  secondaryButtonText: PropTypes.string,
+  secondaryButtons: (props, propName, componentName) => {
+    if (props.secondaryButtons) {
+      if (
+        !Array.isArray(props.secondaryButtons) ||
+        props.secondaryButtons.length !== 2
+      ) {
+        return new Error(
+          `${propName} needs to be an array of two button config objects`
+        );
+      }
+
+      const shape = {
+        buttonText: PropTypes.node,
+        onClick: PropTypes.func,
+      };
+
+      props[propName].forEach((secondaryButton) => {
+        PropTypes.checkPropTypes(
+          shape,
+          secondaryButton,
+          propName,
+          componentName
+        );
+      });
+    }
+
+    return null;
+  },
+  secondaryClassName: PropTypes.string,
+};
+
+export const ModalFooter = React.forwardRef(function ModalFooter(
+  {
+    children,
+    className,
+    closeModal,
+    danger,
+    inputref,
+    onRequestClose,
+    onRequestSubmit,
+    primaryButtonDisabled,
+    primaryButtonText,
+    primaryClassName,
+    secondaryButtonText,
+    secondaryButtons,
+    secondaryClassName,
+    ...other
+  },
+  ref
+) {
+  const prefix = usePrefix();
+
+  const footerClass = classNames({
+    [`${prefix}--modal-footer`]: true,
+    [className]: className,
+    [`${prefix}--modal-footer--three-button`]:
+      Array.isArray(secondaryButtons) && secondaryButtons.length === 2,
+  });
+
+  const primaryClass = classNames({
+    [primaryClassName]: primaryClassName,
+  });
```

Comment:
This is so confusing to me lol, could it just be used directly as `primaryClassName` without ever being passed to `classNames`? 🤔